### DISCUSSION TITLE
[iOS] request subscription with promotional offer

### DISCUSSION THREAD

**Discussion by rebeccahsu** [hyochan/react-native-iap#2568] [Q&A] [answered]
Hi everyone,

I want to implement iOS Promotional Offers with the library. I can get the promotional price (discounts array) in `getSubscriptions()` but when I do `requestPurchase({ sku, withOffer: { identifier: '...', keyIdentifier: '...', nonce: '...', signature: '...', timestamp: '...' } })`, the discount doesn't apply and it shows the normal price for the customer. Does anyone know how do I request the promotional price? What parameters should be passed besides the sku? Thanks!

Btw I've read the [Official Doc of Apple Promotional Offers](https://developer.apple.com/documentation/storekit/in-app_purchase/original_api_for_in-app_purchase/subscriptions_and_offers/implementing_promotional_offers_in_your_app) but still have no idea how to implement this with RNIap.

**Comment by rebeccahsu** [comment, marked as answer]
Found it. should be:
```
RNIap.requestSubscription({
  sku: '...',
  appAccountToken: '...'
  withOffer: {
     identifier: '...',
     keyIdentifier: '...',
     nonce: '...',
     signature: '...',
     timestamp: '...'
  }
})
```

**Reply by itsaliraxa** [reply]
Hi @rebeccahsu I am getting an error while I try to buy 
Error: An unknown error occurred at Object.promiseMethodWrapper [as buyProduct]
